pkg/sql/opt/exec/execbuilder/testdata/scalar

```diff
@@ -1301,3 +1301,13 @@ vectorized: true
       estimated row count: 1,000 (missing stats)
       table: t0@t0_pkey
       spans: FULL SCAN
+
+# Regression test for not checking whether arguments to a binary op are non-NULL
+# when folding (#94264).
+query T
+EXPLAIN SELECT 2-(9223372036854775807+436256318) < (CASE WHEN false THEN -1 END);
+----
+distribution: local
+vectorized: true
+·
+• norows
```

Comment:
It'd be nice to have normalization tests for each of the three rules similar (and right after) each of these tests: 

* https://github.com/cockroachdb/cockroach/blob/91505867f66ee931d37b1e3ddea0fb3937c72dc3/pkg/sql/opt/norm/testdata/rules/comp#L152-L163
* https://github.com/cockroachdb/cockroach/blob/91505867f66ee931d37b1e3ddea0fb3937c72dc3/pkg/sql/opt/norm/testdata/rules/comp#L209-L220
* https://github.com/cockroachdb/cockroach/blob/91505867f66ee931d37b1e3ddea0fb3937c72dc3/pkg/sql/opt/norm/testdata/rules/comp#L267-L282

You can exercise the same bug for `NormalizeCmpMinusConst` with:

```
SELECT (9223372036854775807+436256318)-2 < (CASE WHEN false THEN -1 END)
```

And `NormalizeCmpPlusConst` with:

```
SELECT (9223372036854775807+436256318)+2 < (CASE WHEN false THEN -1 END)
```

Reply:
Sorry for not mentioning this before.